File: apps/web/lib/validations/teacher.ts
import { z } from 'zod';

export const teacherFormSchema = z.object({
  // User fields
  email: z.string().email('Invalid email address'),
  password: z.string().min(6, 'Password must be at least 6 characters').optional(),

  // Personal details
  firstName: z.string().min(2, 'First name must be at least 2 characters'),
  lastName: z.string().min(2, 'Last name must be at least 2 characters'),
  dateOfBirth: z.string(),
  gender: z.enum(['MALE', 'FEMALE', 'OTHER']),
  bloodGroup: z.string().optional(),
  phone: z.string().optional(),
  photo: z.string().optional(),

  // Contact details
  address: z.string().optional(),
  city: z.string().optional(),
  state: z.string().optional(),
  pincode: z.string().optional(),
  emergencyContact: z.string().optional(),

  // Professional details
  qualification: z.string().min(2, 'Qualification is required'),
  specialization: z.string().optional(),
  experience: z.coerce.number().min(0).default(0),
  joiningDate: z.string().optional(),
  departmentId: z.string().optional(),
  designation: z.string().optional(),
  basicSalary: z.coerce.number().min(0, 'Basic salary must be positive'),

  // Status
  status: z.enum(['ACTIVE', 'ON_LEAVE', 'RESIGNED', 'TERMINATED']).default('ACTIVE'),
});

export type TeacherFormValues = z.infer<typeof teacherFormSchema>;

export const attendanceFormSchema = z.object({
  date: z.string(),
  status: z.enum(['PRESENT', 'ABSENT', 'LATE', 'HALF_DAY', 'LEAVE']),
  inTime: z.string().optional(),
  outTime: z.string().optional(),
  remarks: z.string().optional(),
});

export type AttendanceFormValues = z.infer<typeof attendanceFormSchema>;

File: apps/android/core/model/src/main/java/com/edunexus/android/core/model/Teacher.kt
package com.edunexus.android.core.model

import com.edunexus.android.core.model.enums.EmployeeStatus
import com.edunexus.android.core.model.enums.Gender
import kotlinx.serialization.Serializable

@Serializable
data class Teacher(
    val id: String,
    val employeeId: String,
    val userId: String,

    // Personal Details
    val firstName: String,
    val lastName: String,
    val dateOfBirth: String,
    val gender: Gender,
    val bloodGroup: String? = null,
    val photo: String? = null,

    // Contact
    val address: String? = null,
    val city: String? = null,
    val state: String? = null,
    val pincode: String? = null,
    val emergencyContact: String? = null,

    // Professional Details
    val qualification: String,
    val specialization: String? = null,
    val experience: Int = 0,
    val joiningDate: String,
    val departmentId: String? = null,
    val designation: String? = null,

    // Salary Details
    val basicSalary: Double,

    // Status
    val status: EmployeeStatus = EmployeeStatus.ACTIVE,
    val leftDate: String? = null,

    val createdAt: String,
    val updatedAt: String
)
